### PR TITLE
feat(server, web): include pictures of shared albums on map

### PR DESCRIPTION
**Pull request by andreasgerstmayr**
This PR adds a new setting "Include shared albums" to the map settings, to includes pictures of albums shared with the current user on the map.

Related: #3176

### CODE REVIEW

File: server/src/infra/repositories/asset.repository.ts
Comment (andreasgerstmayr) [inline server/src/infra/repositories/asset.repository.ts:514]: Is there a better way to write this OR expression?
`WHERE (ownerID IN ... OR albumId IN ...) AND isVisible = ... AND isArchived = ... AND ...`

File: server/src/domain/asset/asset.service.ts
Comment (jrasm91) [inline server/src/domain/asset/asset.service.ts:173]: This doesn't include the asset relation does it?

File: server/src/domain/asset/dto/map-marker.dto.ts
Comment (jrasm91) [inline server/src/domain/asset/dto/map-marker.dto.ts:35]: This should not be needed

File: server/src/infra/repositories/asset.repository.ts
Comment (jrasm91) [inline server/src/infra/repositories/asset.repository.ts:541]: This section is repeated twice. Can you move it up to a variable?

File: web/src/lib/components/map-page/map-settings-modal.svelte
Comment (jrasm91) [inline web/src/lib/components/map-page/map-settings-modal.svelte:38]: Should we remember this other option so it is more obvious what the difference is?

**Review by jrasm91 — APPROVED** [review]
This looks really good. Thanks!

**Comment by danieldietzler** [comment]
@andreasgerstmayr You've requested a review from me but you haven't yet addressed @jrasm91 comments and there are also lots of merge conflicts right now...?

**Comment by andreasgerstmayr** [comment]
> @andreasgerstmayr You've requested a review from me but you haven't yet addressed @jrasm91 comments and there are also lots of merge conflicts right now...?

hi @danieldietzler, sorry about that, must have been some automated process (it's showing up as 2:35 in the night), maybe because you're listed as a codeowner. No idea why it's happening now and not when I created the PR though.

Sorry for the late response, I was on holiday for a couple of weeks and then didn't get time so far.
Thanks for the review, I'll rebase and implement the changes soon.

**Comment by danieldietzler** [comment]
Ah ok got it. No rush then :)

File: server/src/domain/asset/dto/map-marker.dto.ts
Comment (andreasgerstmayr) [reply server/src/domain/asset/dto/map-marker.dto.ts]: :+1:

The class got moved to `server/src/dtos/search.dto.ts` and doesn't have this annotation anymore.

File: server/src/infra/repositories/asset.repository.ts
Comment (andreasgerstmayr) [reply server/src/infra/repositories/asset.repository.ts]: Good idea, done.

File: web/src/lib/components/map-page/map-settings-modal.svelte
Comment (andreasgerstmayr) [reply web/src/lib/components/map-page/map-settings-modal.svelte]: I renamed `Include shared with me` to `Include shared from partners`. Another option would be `Include assets from partners`.

What do you think, which option sounds better? Or do you have other suggestions?

File: server/src/domain/asset/asset.service.ts
Comment (andreasgerstmayr) [reply server/src/domain/asset/asset.service.ts]: I don't follow, where should I include the asset relation?

**Comment by andreasgerstmayr** [comment]
I fixed the merge conflicts and implemented (most of) @jrasm91 suggestions now (except the one I didn't understand, see above).
Sorry for the delay on my side!

File: web/src/lib/components/map-page/map-settings-modal.svelte
Comment (NicholasFlamy) [reply web/src/lib/components/map-page/map-settings-modal.svelte]: I'm not Jason but I would suggest `Include partner assets` or `Include shared partner assets`. Also, this rename is important for clarity.

Comment (andreasgerstmayr) [reply web/src/lib/components/map-page/map-settings-modal.svelte]: I renamed it to `Include shared partner assets` now, thanks for the suggestion @NicholasFlamy

File: server/src/domain/asset/asset.service.ts
Comment (NicholasFlamy) [reply server/src/domain/asset/asset.service.ts]: Okay so Jason answered me on Discord about this:
https://discord.com/channels/979116623879368755/994044917355663450/1238973376349864108

Jason said: "Just want to make sure this is not loading the album AND all the assets for it"

I said: "If I understand correctly, you are asking if it is loading only the shared albums when figuring out what to show on the map, versus loding the shared albums and all the assets. And the problem with that is loading all the assets in the shared album takes more resources."

Jason responded: "Yup"

Comment (andreasgerstmayr) [reply server/src/domain/asset/asset.service.ts]: yes, this function does not query all assets, only the albums:
https://github.com/immich-app/immich/blob/f667c9597b9885059de00d9e8779147c4d08ddc2/server/src/repositories/album.repository.ts#L154-L166

**Review by zackpollard — APPROVED** [review]
LGTM, will let @jrasm91 take a look over it before we merge though

**Comment by zackpollard** [comment]
I fixed a tsc error caused by changes in main since the branch was created as well as a prettier error in web. The PR should now be good to merge

**Review by jrasm91 — APPROVED** [review]
I updated this to include assets that are in shared albums where the user is the owner.

**Comment by andreasgerstmayr** [comment]
> I updated this to include assets that are in shared albums where the user is the owner.

Good catch! I didn't think of that.

Thank you for merging! :)

**Pull request merged**
